pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,295 @@
+package grouppermission
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/openshift/rbac-permissions-operator/pkg/apis"
+	"github.com/openshift/rbac-permissions-operator/pkg/apis/managed/v1alpha1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// create fake client to mock API calls
+func newTestReconciler() *ReconcileGroupPermission {
+	return &ReconcileGroupPermission{
+		client: fake.NewFakeClient(),
+		scheme: scheme.Scheme,
+	}
+}
+
+// create a GroupPermission object so we can resigter it in the fake client
+func mockGroupPermission() *v1alpha1.GroupPermission {
+	return &v1alpha1.GroupPermission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testGroupPermission",
+			Namespace: "rbac-permissions-operator",
+		},
+		Spec: v1alpha1.GroupPermissionSpec{
+			GroupName: "exampleGroupName",
+			ClusterPermissions: []v1alpha1.ClusterPermission{
```

Comment:
`v1alpha1.ClusterPermission` not defined.

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_rbac-permissions-operator/12/pull-ci-openshift-rbac-permissions-operator-master-images/19